### PR TITLE
Add `cirq.two_qubit_matrix_to_cz_isometry` decomposition for single to two qubit isometry using at-most 2CZs.

### PR DESCRIPTION
**Pull request by tanujkhattar**
* Given an arbitrary 2-qubit unitary $U$; we can decompose it into at-most 3 CZs + single qubit rotations. 
* However, if we know that one of the 2 input qubits is always in state |0>; we can implement the action of unitary $U$ using at-most 2 CZs + single qubit rotations. 
* `cirq.two_qubit_matrix_to_cz_isometry` implements this decomposition using `https://arxiv.org/abs/1501.06911` as a reference.

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
For my own reference: Appendix B of the linked paper describes the one-to-two qubit isometry (i.e. a two-qubit gate which takes a `|0)` state as one of its inputs).

File: cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry_test.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    'initial_state, is_valid',
+    list(
+        zip(
+            VALID_INITIAL_STATES + INVALID_INITIAL_STATES,
+            [True] * len(VALID_INITIAL_STATES) + [False] * len(INVALID_INITIAL_STATES),
+        )
+    ),
+)
+@pytest.mark.parametrize('unitary_matrix', [cirq.testing.random_unitary(4) for _ in range(10)])
+@pytest.mark.parametrize('allow_partial_czs', [True, False])
```
Comment (95-martin-orion) [inline cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry_test.py:41]: These will generate _160_ test cases, which seems excessive. Is there a subset we can reduce this to?

File: cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        operations and ejecting phased Paulis and Z operations.
+
+    Returns:
+        A list of operations implementing the action of the given unitary matrix, assuming
+        the input qubit `q0` is in the |0> state.
+    """
+    d, cz_ops = two_qubit_to_cz.two_qubit_matrix_to_diagonal_and_cz_operations(
+        q0, q1, mat, allow_partial_czs, atol, clean_operations
+    )
+    decomposed_ops = [ops.PhasedXZGate.from_matrix(np.diag([d[0][0], d[1][1]])).on(q1)] + cz_ops
```
Comment (95-martin-orion) [inline cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry.py:56]: [No change needed] I take it this is the "replace the diagonal gate ∆ of the last circuit by a single-qubit diagonal gate acting on the least significant qubit" step described in the paper?

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]

File: cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry.py (hunk 2 above)
Comment (tanujkhattar) [reply cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry.py:56]: Yes, that's correct.

File: cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry_test.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry_test.py]: Reduced the number of test cases to 60 instead.

**Review by 95-martin-orion — APPROVED** [review]

**Pull request merged**